### PR TITLE
Enable goimports

### PR DESCRIPTION
**Pull request by shuheiktgw**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Enable goimports and automatically detects unorganized imports

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes-sigs/kueue/pull/228#discussion_r856681587

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    - errcheck
+    - goimports
+    - gosimple
+    - govet
+    - ineffassign
+    - staticcheck
+    - structcheck
+    - typecheck
+    - unused
+    - varcheck
```
Comment (shuheiktgw) [inline .golangci.yaml:29]: These linters are automatically enabled by default except for goimports, so the actual change is simply enabling it
- https://golangci-lint.run/usage/linters/#enabled-by-default

File: pkg/controller/core/clusterqueue_controller.go [hunk 2]
```diff
@@ -20,7 +20,6 @@ import (
 	"context"
 
 	"github.com/go-logr/logr"
-	"sigs.k8s.io/kueue/pkg/constants"
```
Comment (alculquicondor) [inline pkg/controller/core/clusterqueue_controller.go:23]: Remove this empty line. let's not have 4 blocks of imports, just 3.

File: test/integration/controller/core/queue_controller_test.go [hunk 3]
```diff
@@ -19,10 +19,11 @@ package core
 import (
 	"github.com/onsi/ginkgo/v2"
 	"github.com/onsi/gomega"
+
```
Comment (alculquicondor) [inline test/integration/controller/core/queue_controller_test.go:22]: remove

File: test/integration/controller/core/workload_controller_test.go [hunk 4]
```diff
@@ -21,10 +21,11 @@ import (
 
 	"github.com/onsi/ginkgo/v2"
 	"github.com/onsi/gomega"
+
```
Comment (alculquicondor) [inline test/integration/controller/core/workload_controller_test.go:24]: remove

**Comment by shuheiktgw** [comment]
@alculquicondor I've fixed the PR so would you take a look at it again? 🙏

**Review by alculquicondor — COMMENTED** [review]
/approve

Can you squash?

**Comment by shuheiktgw** [comment]
@alculquicondor Thank you for your review! I've squashed the commits so would you check the PR again? 🙂

**Pull request merged**